bin/yang_get_stats.py

```diff
@@ -441,43 +72,383 @@ def write_dictionary_file_in_json(in_dict, path, file_name):
-
-    print()
-    print('Print, per company, the number of IETF drafts containing YANG model(s)')
-    print('Total numbers of drafts with YANG Model(s): {} - non strict rules: {}'
-          .format(total_number_drafts, total_number_drafts_no_strict))
-    print()
-
-    for company in companies:
-        print_attribution(*company)
-
-    for f in remove_old_html_files:
-        try:
-            os.unlink(f)
-        except:
-            # ignore if the file does not exist
-            pass
-
-    # diff between files and files_no_strict lists
-    files_diff = []
-    for f in files_no_strict:
-        if f not in files:
-            files_diff.append(f)
-            bash_command = "cp " + draft_path_nostrict + f + " " + draft_path_diff
+            self.files = all_files
+        self.remove_old_html_files = []
+
+        self.gather_yang_page_main_compilation_stats()
+        self.gather_ietf_yang_page_main_compilation_stats()
+        self.gather_backups_compilation_stats()
+        self.gather_ietf_yang_out_of_rfc_compilation_stats()
+
+        self.print_files_information()
+
+        for filename in self.remove_old_html_files:
+            try:
+                os.unlink(filename)
+            except FileNotFoundError:
+                pass
+
+    def gather_yang_page_main_compilation_stats(self):
+        json_history_file = os.path.join(self.backup_directory, f'{self.YANG_PAGE_MAIN_PREFIX}history.json')
+        yang_page_compilation_stats = self._load_compilation_stats_from_history_file(json_history_file)
+        for filename in self._file_names_containing_keyword(self.files, self.YANG_PAGE_MAIN_PREFIX):
+            generated_at = 0
+            passed = 0
+            passed_with_warnings = 0
+            failed = 0
+            extracted_date = self._extract_date_from_filename(filename)
+            if (datetime.date.today() - extracted_date).days > 30:
+                self.remove_old_html_files.append(self.backup_directory + filename)
+            i = 0
+            with open(self.backup_directory + filename) as f:
+                for line in f:
+                    if i == 1:
+                        generated_at = line.split('on')[-1].split('by')[0].strip()
+                    elif i == 5:
+                        passed = int(line.split(':')[-1].split('/')[0])
+                    elif i == 6:
+                        passed_with_warnings = int(line.split(':')[-1].split('/')[0])
+                    elif i == 7:
+                        failed = int(line.split(':')[-1].split('/')[0])
+                    elif i == 8:
+                        i = 0
+                        yang_page_compilation_stats[date2num(extracted_date)] = {
+                            'name': {
+                                'generated-at': generated_at,
+                                'passed': passed,
+                                'warnings': passed_with_warnings,
+                                'failed': failed
```

Comment:
What about directly creating dict, which will group these 4 integer values and it will be assigned to the `name`
```
yang_page_compilation_stats[date2num(extracted_date)] = {
    'name': results
}
```
NOTE: If you will make this change, do it also in other methods

Reply:
I don't really think it will add readability to this code part

Reply:
Ok, no problem